sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/LocalHttpClient.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+package com.azure.containers.containerregistry;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.test.http.MockHttpResponse;
+import com.azure.core.test.models.NetworkCallRecord;
+import com.azure.core.test.models.RecordedData;
+import com.azure.core.util.Context;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.Map;
+
+public class LocalHttpClient implements HttpClient {
```

Comment:
I'm trying to understand the purpose of this class, is this supposed to be for scenarios where a testing recording was created against a canary-like environment that isn't available when running integration tests against live Azure resources? If this is the case, just have the tests use `PlaybackClient` as this class effectively emulates that without the additional guards there.

Reply:
The ACR service does not support classic CRUD operations for v1 endpoints which is what we are supported currently through our library - create is supported via either docker or ARM APIs . 

Also, the create that is supported via docker\ARM Apis is create of an entire image - which is grouped bundle of manifests, tags etc, is super bulky etc.). They also support static caching on operations (no invalidation on update/delete) - so frequent delete\create\update can cause issues as they don't expect people to do this - but we need to do it as part of our testing.

Given the above we have following options
1. Any update\delete needs to be done on a different image - Did not choose this option since it is too costly with our matrix of each method becomes 4 methods. and each functionality is supported on hierarchial elements - so repository, tag and manifest. Also manifest operations are supported on both parent manifest and child manifest.

2. Reduce the matrix with minimal throw away code. This state is temporary as we will be supporting v2 endpoints and will be able to create idempotent tests and the service plans to remove the caching, since dates are still up in the air did not want to ignore test coverage.

1. I have created 3 integration test classes
a. Get operations on registry async client via WIthreponse methods.
b. Get operations on repository async client via WithReponse methods.
c. Write operations on the repository async clients via WithResponse which run synchronously since we are dealing with a single image and has time delays to overcome the cache.
d. I still want to add coverage for the other 3 permutatations of each method and for that I add my own HttpClient that instead of calling the service uses the record APIs from the WithResponse calls. This is the easiest way to get the same coverage as-if I was calling the service. As once the pending items are done -  I will only need to change the HttpClient and everything will work as normal. 

Let me know if you have any questions\concerns.